cohortextractor/study_definition.py

```diff
@@ -71,6 +72,7 @@ def to_file(
             )
             dataframe_to_file(df, filename)
         elif dummy_data_file:
+            validate_dummy_data(self, dummy_data_file)
```

Comment:
I think we need something to check that the dummy data file format matches the requested output format. That's almost as simple as checking that they have the same extension although the two level extensions (e.g. `.csv.gz`) make that a bit more tricky.